homeassistant/components/epic_games_store/config_flow.py

```diff
@@ -0,0 +1,95 @@
+"""Config flow for Epic Games Store integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from epicstore_api import EpicGamesStoreAPI
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from . import get_country_from_locale
+from .const import CONF_LOCALE, CONF_SUPPORTED_LOCALES, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_LOCALE): vol.In(CONF_SUPPORTED_LOCALES),
+    }
+)
+
+
+def get_default_locale(hass: HomeAssistant) -> str:
+    """Get default locale code based on Home Assistant config."""
+    locale_code = f"{hass.config.language}-{hass.config.country}"
+    if locale_code in CONF_SUPPORTED_LOCALES:
+        return locale_code
+    if hass.config.language in CONF_SUPPORTED_LOCALES:
+        return hass.config.language
+    return "en-US"
+
+
+async def validate_input(
+    hass: HomeAssistant, user_input: dict[str, Any]
+) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+    api = EpicGamesStoreAPI(
+        user_input[CONF_LOCALE], get_country_from_locale(user_input[CONF_LOCALE])
+    )
+    data = await hass.async_add_executor_job(api.get_free_games)
+
+    if data.get("errors"):
+        _LOGGER.error(data["errors"])
+        raise CannotConnect
+
+    # Return info that you want to store in the config entry.
+    return {
+        CONF_LOCALE: user_input[CONF_LOCALE],
+    }
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Epic Games Store."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        data_schema = self.add_suggested_values_to_schema(
+            STEP_USER_DATA_SCHEMA,
+            user_input or {CONF_LOCALE: get_default_locale(self.hass)},
+        )
+        if user_input is None:
```

Comment:
This is personal preference: I like to have my config flow steps inverted. So I do validation etc within the if. The biggest pro is that you only need one `async_show_form` (there are exceptions of course). LastFM has a good example

Reply:
Why not, but I kinda get tired of editing the PR every month for few things.

The config flow is nice now, I understand your point but could be seen later on.

I am thinking of redoing all config flows of all my integrations I get your comment in mind.


I think we can leave it that way for now and follow with something better afterwards.